### DISCUSSION TITLE
How to Avoid Emborg Recopying Existing Files From Archive Created Directly through Borg

### DISCUSSION THREAD

**Discussion by akibuflash** [KenKundert/emborg#68] [Q&A]
I have an existing repository and archive consisting of several terabytes of data that I created directly using Borg (e.g. borg create ..., etc.).  I now want to experiment with using emborg to backup but I don't want to have emborg copy over the many terabytes of data.  Will emborg recopy the files upon its first run or will it automatically know that those files already exist in the repository and merely only copy those files which have been changed since my original borg backup.  Thanks in advance.

Akibu

**Comment by KenKundert** [comment]
No, Emborg won't re-copy your files.  Borg is de-duplicating, so even if you had the same files on a completely different machine and backed them up to the same repository, borg would not rec-copy the files.  When borg encounters a file, it computes the signature of the contents and compares it to signatures of the files already copied.  It will not copy the file again if there is already a copy in the repository.  The fact that emborg runs borg for you rather than you running borg directly will not change that.

If I were you I would copy your borg settings into the emborg config files, and then run experiments to make sure everything is configured propertly before I used emborg to back up you files.  For example, you can use emborg to list your archives, or list the files in an archive, or you can run the info command to get information about your repository, or you can mount an archive, etc.  All of these things do not actually modify your repository, and should help you gain confidence that things are set up properly.

When it comes time to try a backup, if things do go awry, you can always simply kill emborg and then delete the most recent archive and then run compact to eliminate any files you did not mean to backup.

If you are still worried, you can experiment with emborg while avoiding your existing borg repository altogether.  Simply create a local repository in /tmp, and backup a small directory to it.  Try all of your experiments.  When you are done experimenting, you can simply delete the repository.  You can examine the emborg log file to see exactly which borg commands emborg is running, so you can use that to help you gain confidence in emborg.

**Comment by akibuflash** [comment]
Thanks @KenKundert!!  I will run some tests and report back.  

> If I were you I would copy your borg settings into the emborg config files, and then run experiments to make sure everything is configured propertly before I used emborg to back up you files.

With respect to copying the borg settings into emborg.  Where are borg settings located and where do I copy them to?  In addition, I assume that there is also some sort of cache file in borg that would tell emborg what has already been done... where would that file be placed in emborg?  Thanks again.

**Reply by KenKundert** [reply]
Borg does not have any kind of configuration file like emborg, so you have to specify the options on the command line when you run borg.  This results in long command lines, so most people create shell scripts to run borg.  You would transfer the command line arguments in your borg script to emborgs settings file.  There is a whole section on configuring in the emborg documentation (emborg.readthedocs.io).

There is a borg configuration directory ($HOME/.config/borg) and a borg cache directory ($HOME/.cache/borg), but there is nothing in either that users need to be aware of.
